functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,276 @@ export const processNewAllotment = functions.database
+
+                    // if this is the first occurence of the file, set its value to the list of chunks 
+                    else    
+                        chunks[currentFile.file_name] = currentFile.chunks;
+                }
+
+                currentFile.file_name = AudioFileName;
+                currentFile.chunks = [chunkToStore];
+            }                    
+        }
+        
+
+        ////////////////////////////////
+        // Processing Sheet Audio Files
+        ////////////////////////////////
+
+        Object.keys(chunks).forEach(async fileName => {
+            let currentFilechunks = chunks[fileName];
+
+            // Sorting by `beginning` time
+            currentFilechunks = await lodash.orderBy(currentFilechunks, ['beginning'], ['asc']);
+            // console.log(currentFilechunks);
+
+            let chunksToImport = [];
+            ////////////////////////////////////////////
+            // [CHECK #5] Chunks of a single audio file 
+            // should not overlap each other in timing.
+            ////////////////////////////////////////////
+            let lastEndingTime = null;
+            currentFilechunks.forEach((chunk, i) => {
+
+
+                // `Beginning` value should be less than `Ending`.
+                if (chunk.beginning > chunk.ending) {
+                    console.log(`Audio file name: ${fileName} -- Error: "Beginning" value isn't less than "Ending".`);
+                    summary.overlappingChunks.push({
+                        audioFileName: fileName,
+                        beginning: chunk.beginning,
+                        ending: chunk.ending,
+                        errorMessage: `"Beginning" value isn't less than "Ending".`
+                    });
+                }
+
+                // `beginning` of the CURRENT chunk should be GREATER than the `ending` of the LAST chunk
+                if (i !== 0 && chunk.beginning && lastEndingTime && chunk.beginning < lastEndingTime) {
+                    console.log(`Audio file name: ${fileName} -- Error: Overlapping chunks.`);
+                    summary.overlappingChunks.push({
+                        audioFileName: fileName,
+                        errorMessage: `Overlapping chunks.`
+                    });
+                }
+
+
+                // setting the value of `lastEndingTime` to be used in the next iteration
+                if (chunk.ending)
+                    lastEndingTime = chunk.ending;
+                else
+                    lastEndingTime = null;
+
+
+                //////////////////////////////////////////////////
+                // [CHECK #4] `continuationFrom` column should
+                // be filled only for the first chunk of the track.
+                //////////////////////////////////////////////////
+                if (chunk.continuationFrom && i !== 0)
+                    delete chunk.continuationFrom;
```

Comment:
You should not modify the data! “Should” means that if it if filled for any other chunk than for the first one, you should issue a warning in the summary email.